net/dnscache/messagecache_test.go

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dnscache
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"golang.org/x/net/dns/dnsmessage"
+)
+
+func TestMessageCache(t *testing.T) {
+	var (
+		mu  sync.Mutex
+		now = time.Date(1987, 11, 1, 0, 0, 0, 0, time.UTC)
+	)
+	sleep := func(sec int) {
+		mu.Lock()
+		defer mu.Unlock()
+		now = now.Add(time.Duration(sec) * time.Second)
+	}
+	mc := &MessageCache{
+		Clock: func() time.Time {
```

Comment:
Use `tstest.Clock`? This is basically doing the same thing but that way you don't need to reimplement it.

Reply:
Done.